src/configure/read_url.rs

```diff
@@ -12,7 +12,7 @@ pub async fn read_url(context: &Context, url: &str) -> Result<String, Error> {
     match surf::get(url).recv_string().await {
         Ok(res) => Ok(res),
         Err(err) => {
-            info!(context, "Can\'t read URL {}", url);
+            info!(context, "Can\'t read URL {}: {}", url, err);
```

Comment:
Here in an ideal world, we wouldn't do any error logging and just report it up.  On line 6 the cause should have a `#[from]` annotation so that you can just propagate using `?` and be done.  And someone much closer to the application should be doing the logging.

I think as a general rule "much closer to the application" for us means the FFI interface should be doing all the error logging.  The rest of the deltachat core should be treated as a library and errors should be propagated out and not logged (bearing in mind there are always exceptions and practical obstacles.  I'm well aware we are far away from being in that nice place)

Reply:
Problem is, this is during configure and many URLs are tried for autoconfig and if we do not find any, we just guess the config. So, if configuration fails, we don't know if the problem was that we could not read the autoconfig or that we could not reach the server. But we can return only one error. Of course we could add more info to this error (which is also shown to the user) but I don't really think that's worth it.